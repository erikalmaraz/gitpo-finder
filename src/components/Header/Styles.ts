import React from "react";
import styled from "styled-components";

export const Wrapper = styled.section`
  display: flex;
  justify-content: space-between;
  min-height: 70px;
  align-items: center;
  padding: 0 50px;
  z-index: 9;
  position: relative;
  img {
    &:nth-child(1) {
      width: 110px;
    }
    &:nth-child(2) {
      width: 30px;
    }
  }
`;

export const AuthorContainer = styled.div`
  display: flex;
  align-items: center;
  font-size: 13px;
  span {
    margin-right: 12px;
  }
`;

export const ThemeControllerWrapper = styled.div`
  display: flex;
  align-items: center;
`;

export const ToggleSwitchContainer = styled.div`
  width: 50px;
  height: 19px;
  background: ${({ theme }) => (theme === "light" ? "#fff" : "#000")};
  border: 2px solid #ddd;
  border-radius: 21px;
  position: relative;
  margin-left: 20px;
  align-items: center;
  cursor: pointer;
`;

export const CircleToggleSwitch = styled.span`
  height: 13px;
  width: 13px;
  background: ${({ theme }) => (theme === "light" ? "#444" : "#fff")};
  display: block;
  border-radius: 100%;
  position: absolute;
  bottom: 1px;
  left: ${({ theme }) => (theme === "light" ? "2px" : "30px")};
  transition: all 0.3s ease;
`;
