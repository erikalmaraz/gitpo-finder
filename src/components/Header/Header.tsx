import React, { useState } from "react";
import HeaderType from "../../interfaces/header.interfaces";
import * as S from "./Styles";
const Header = ({ themeToggler }: HeaderType) => {
  const [theme, setTheme] = useState("light");

  const toggleTheme = () => {
    const isLigthThemeSelected = theme === "light";
    themeToggler();
    if (isLigthThemeSelected) {
      setTheme("dark");
    } else {
      setTheme("light");
    }
  };

  return (
    <React.Fragment>
      <S.Wrapper>
        <div>
          <img src="../assets/images/logo-01.png" alt="Logo" />
        </div>
        <S.ThemeControllerWrapper>
          <S.AuthorContainer>
            <span>By Erik Almaraz</span>
            <img src="../assets/icons/github-icon.png" alt="Github repo" />
          </S.AuthorContainer>
          <S.ToggleSwitchContainer onClick={toggleTheme} theme={theme}>
            <S.CircleToggleSwitch theme={theme}></S.CircleToggleSwitch>
          </S.ToggleSwitchContainer>
        </S.ThemeControllerWrapper>
      </S.Wrapper>
    </React.Fragment>
  );
};

export default Header;
