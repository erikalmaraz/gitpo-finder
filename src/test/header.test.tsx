import React from "react";
import { render, screen } from "@testing-library/react";
import Header from "../components/Header/Header";
import renderer from "react-test-renderer";

it("Should render Loader correctly", () => {
  const props = {
    themeToggler: jest.fn,
  };
  const tree = renderer.create(<Header {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
