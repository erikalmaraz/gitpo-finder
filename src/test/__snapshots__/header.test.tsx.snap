// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Loader correctly 1`] = `
<section
  className="sc-bdfBQB joztvB"
>
  <div>
    <a
      aria-current="page"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Logo"
        src="../assets/images/logo-01.png"
      />
    </a>
  </div>
  <div
    className="sc-dlfnuX hnnDGO"
  >
    <div
      className="sc-gsTEea bPVsIV"
    >
      <span>
        By Erik Almaraz
      </span>
      <img
        alt="Github repo"
        src="../assets/icons/github-icon.png"
      />
    </div>
    <div
      className="sc-hKgJUU bzhxfT"
      onClick={[Function]}
    >
      <span
        className="sc-eCstlR jEXLEy"
      />
    </div>
  </div>
</section>
`;
